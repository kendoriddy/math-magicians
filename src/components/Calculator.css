:root {
  --primary: rgb(153, 152, 152);
  --digit: #f5f5f5;
  --operator: rgb(255, 144, 104);
}

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.overall {
  width: 100%;
  max-width: 400px;
  margin: 50px auto;
  background-color: var(--primary);
}

.result-span {
  background-color: var(--primary);
  text-align: right;
  padding: 1.5rem;
  color: var(--digit);
}

.grid-one,
.grid-two,
.grid-three,
.grid-four,
.grid-five {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

button {
  border: none;
  outline: none;
  font-size: 1rem;
  appearance: none;
  margin-right: 1rem;
  background: none;
  cursor: pointer;
  transition: all 0.2s ease-in-out;
}

.grid-one button,
.grid-two button,
.grid-three button,
.grid-four button {
  margin: 1px 1px 1px 0;
  padding: 0.5rem;
}

.zero {
  grid-column: 1 / span 2;
}

.grid-five button {
  margin: 1px 1px 1px 0;
  padding: 0.5rem;
}

.operator,
.number,
.empty {
  border-width: 0;
  font-weight: bold;
  font-size: 25px;
  cursor: pointer;
}

.empty {
  background-color: var(--digit);
}

.number {
  background-color: var(--digit);
}

.operator {
  background-color: var(--operator);
  font-size: 30px;
}
