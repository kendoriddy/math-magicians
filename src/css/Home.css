.home-overall h1 {
  font-size: 2rem;
  font-weight: bold;
  color: #595252;
  padding: 1.5rem 3rem;
}

.home-overall p {
  padding: 1rem 3rem;
  font-size: 1rem;
  color: #292626;
}
