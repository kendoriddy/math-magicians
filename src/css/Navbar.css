.navbar {
  padding: 0.5rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-size: 1rem;
  font-weight: bold;
  color: #f2f2f2;
  background-color: rgb(197, 191, 191);
}

.navbar a {
  text-decoration: none;
  padding: 6px;
  color: #f2f2f2;
  list-style: none;
}

.nav-items {
  display: flex;
  justify-content: space-between;
}

.nav-items li {
  list-style: none;
}

/* .navbar .active {
  font-weight: 600;
  border-bottom: 2px solid var(--orange);
} */
