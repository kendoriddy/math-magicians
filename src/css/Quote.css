.quote-container {
  width: 90%;
  max-width: 400px;
  min-height: 280px;
  padding: 2.5rem;
  margin: 8rem auto;
  background: #030413;
  border-radius: 4px;
  box-shadow: 4px 4px 1px rgba(0, 0, 0, 0.5);
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: flex-start;
}

p {
  color: #f2f2f2;
  font-size: 1.5rem;
  word-wrap: break-word;
}

p:last-of-type {
  margin: 0.8rem 0;
}

.author {
  font-weight: bold;
}

a {
  text-decoration: none;
  display: inline-block;
  text-align: center;
}

.buttons-div {
  display: flex;
  justify-content: space-between;
}

.buttons {
  background-color: #d29720;
  color: #f2f2f2;
  display: flex;
  justify-content: space-around;
  padding: 0.5rem 0;
  width: 100px;
  font-size: 14px;
  border-radius: 4px;
  cursor: pointer;
  transition: all 0.3s ease-in-out;
}

.quote-btn {
  border: none;
  outline: none;
  padding: 0.6rem;
  margin-right: 0.9rem;
}

.buttons:hover {
  background-color: #5a57ec;
}
