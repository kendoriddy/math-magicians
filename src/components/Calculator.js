import React, { useState } from "react";
import calculate from "../logic/calculate";
import "../css/Calculator.css";

const Calculator = () => {
  const [input, setInput] = useState({
    total: 0,
      next: null,
      operator: null,
  });
  const handleClick = (e) => {
    const result = calculate(input, e.target.textContent);
    setInput(result);
  };

  const resultDisplayClick = (e) => {
    setInput({
      total: e.target.textContent,
    });
  };

      const { total, next, operation } = input;
    return (
      <div className="overall">
        <div className="result-span" onChange={resultDisplayClick}>
            {total}
            {operation}
            {next}
        </div>
        <div className="grid-one">
          <button onClick={handleClick} type="button" className="btn delete number">
            AC
          </button>
          <button onClick={handleClick} type="button" className="btn number number">
            +/-
          </button>
          <button onClick={handleClick} type="button" className="btn clear number">
            %
          </button>
          <button onClick={handleClick} type="button" className="btn  equal-to operator">
            รท
          </button>
        </div>
        <div className="grid-two">
          <button onClick={handleClick} type="button" className="btn number">
            7
          </button>
          <button onClick={handleClick} type="button" className="btn number">
            8
          </button>
          <button onClick={handleClick} type="button" className="btn number">
            9
          </button>
          <button onClick={handleClick} type="button" className="btn number operator">
            x
          </button>
        </div>
        <div className="grid-three">
          <button onClick={handleClick} type="button" className="btn number">
            4
          </button>
          <button onClick={handleClick} type="button" className="btn number">
            5
          </button>
          <button onClick={handleClick} type="button" className="btn number">
            6
          </button>
          <button onClick={handleClick} type="button" className="btn number  operator">
            -
          </button>
        </div>
        <div className="grid-four">
          <button onClick={handleClick} type="button" className="btn number">
            1
          </button>
          <button onClick={handleClick} type="button" className="btn number">
            2
          </button>
          <button onClick={handleClick} type="button" className="btn number">
            3
          </button>
          <button onClick={handleClick} type="button" className="btn number operator">
            +
          </button>
        </div>
        <div className="grid-five">
          <button onClick={handleClick} type="button" className="btn number zero">
            0
          </button>
          <button onClick={handleClick} type="button" className="btn number dot">
            .
          </button>

          <button onClick={handleClick} type="button" className="btn number operator">
            =
          </button>
        </div>
      </div>
    );
}

export default Calculator;
