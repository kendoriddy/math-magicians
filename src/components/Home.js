import React from "react";
import '../css/Home.css';

const Home = () => {
  return (
    <div className="home-overall">
      <h1>Welcome! You are celebrated</h1>
      <p>
        Mathematics is the science and study of quality, structure, space, and change.
        Mathematicians seek out patterns, formulate new conjectures, and establish truth by rigorous
        deduction from appropriately chosen axioms and definitions.
      </p>

      <p>
      Since the beginning of recorded history, mathematical discovery has been at the forefront of every civilized society, and math has been used by even the most primitive and earliest cultures. The need for math arose because of the increasingly complex demands from societies around the world, which required more advanced mathematical solutions.
      </p>
    </div>
  );
};

export default Home;
