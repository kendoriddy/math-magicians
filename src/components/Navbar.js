import { NavLink } from "react-router-dom";
import "../css/Navbar.css";

const Navbar = () => {
  const links = [
    {
      id: 1,
      path: "/",
      text: "Home",
    },
    {
      id: 2,
      path: "/calculate",
      text: "Calculator",
    },

    {
      id: 3,
      path: "/roman",
      text: "Roman",
    },
    {
      id: 4,
      path: "/quotes",
      text: "Quotes",
    },
  ];

  return (
    <div className="navbar">
      <h2>Math Magician</h2>
      <ul className="nav-items">
        {links.map((link) => (
          <li key={link.id}>
            <NavLink
              to={link.path}
              style={({ isActive }) => ({
                color: isActive ? "#fe9167" : "white",
                textDecoration: isActive ? "underline" : null,
              })} end
            >
              {link.text}
            </NavLink>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Navbar;
