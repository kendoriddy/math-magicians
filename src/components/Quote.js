import React, { useState, useEffect, useRef } from "react";
import '../css/Quote.css';

const Quote = () => {
  const [quotes, setQuotes] = useState("");
  const textRef = useRef();
  const colors = ["#90d000", "#ad5622", "#ea23ff", "#cc0f5d", "#51abce", "#21edac"];

  const getQuotes = () => {
    fetch("https://random-math-quote-api.herokuapp.com/")
      .then((res) => res.json())
      .then((data) => {
        setQuotes(data);
      });
  };

  useEffect(() => {
    getQuotes();
  }, []);

  useEffect(() => {
    // textRef.current.style.color = colors[Math.floor(Math.random() * colors.length)];
  }, [quotes]);
  return (
    <>
      <div className='quote-container'>
      <p ref={textRef}>{quotes.quote}</p>
      <p className='author'>{quotes.author}</p>
    <div className="buttons-div">
          <button className='buttons quote-btn' onClick={getQuotes}>Get Quotes</button>
          <a href={`https://twitter.com/intent/tweet?text=${quotes.quote}`} target='_blank'
            rel='noopener noreferrer'
          className='buttons'>Tweet</a>
        </div>
        </div>
    </>
  );
};

export default Quote;
