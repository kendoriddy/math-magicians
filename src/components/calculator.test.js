import React from 'react';
import renderer from 'react-test-renderer';
import { Router } from 'react-router-dom';

import Home from './Home';
import Quote from './Quote';
import Navbar from './Navbar';
import Calculator from './Calculator';

describe('Component test', () => {
  it('Should render the home component', () => {
    const home = renderer.create(<Home />).toJSON();
    expect(home).toMatchSnapshot();
  });

  it('Should render the quote component', () => {
    const quote = renderer.create(<Quote />).toJSON();
    expect(quote).toMatchSnapshot();
  });

  it('Should render the calculator component', () => {
    const calculator = renderer.create(<Calculator />).toJSON();
    expect(calculator).toMatchSnapshot();
  });

  it('Should render the navbar component', () => {
    <Router>
      const navbar = renderer.create(
      <Navbar />
      ).toJSON(); expect(navbar).toMatchSnapshot();
    </Router>;
  });
});