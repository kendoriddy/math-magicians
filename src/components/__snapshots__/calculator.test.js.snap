// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component test Should render the calculator component 1`] = `
<div
  className="overall"
>
  <div
    className="result-span"
    onChange={[Function]}
  >
    0
  </div>
  <div
    className="grid-one"
  >
    <button
      className="btn delete number"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="btn number number"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="btn clear number"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn  equal-to operator"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="grid-two"
  >
    <button
      className="btn number"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn number"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn number"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn number operator"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="grid-three"
  >
    <button
      className="btn number"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn number"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn number"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn number  operator"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="grid-four"
  >
    <button
      className="btn number"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn number"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn number"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn number operator"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="grid-five"
  >
    <button
      className="btn number zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn number dot"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn number operator"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Component test Should render the home component 1`] = `
<div
  className="home-overall"
>
  <h1>
    Welcome! You are celebrated
  </h1>
  <p>
    Mathematics is the science and study of quality, structure, space, and change. Mathematicians seek out patterns, formulate new conjectures, and establish truth by rigorous deduction from appropriately chosen axioms and definitions.
  </p>
  <p>
    Since the beginning of recorded history, mathematical discovery has been at the forefront of every civilized society, and math has been used by even the most primitive and earliest cultures. The need for math arose because of the increasingly complex demands from societies around the world, which required more advanced mathematical solutions.
  </p>
</div>
`;

exports[`Component test Should render the quote component 1`] = `
<div
  className="quote-container"
>
  <p />
  <p
    className="author"
  />
  <div
    className="buttons-div"
  >
    <button
      className="buttons quote-btn"
      onClick={[Function]}
    >
      Get Quotes
    </button>
    <a
      className="buttons"
      href="https://twitter.com/intent/tweet?text=undefined"
      rel="noopener noreferrer"
      target="_blank"
    >
      Tweet
    </a>
  </div>
</div>
`;
